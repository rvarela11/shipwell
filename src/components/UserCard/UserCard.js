/* eslint-disable camelcase */
// @vendors
import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

// @material-ui
import { withStyles } from '@material-ui/core/styles';
import Card from '@material-ui/core/Card';
import CardHeader from '@material-ui/core/CardHeader';
import Avatar from '@material-ui/core/Avatar';
import CardContent from '@material-ui/core/CardContent';

// @styles
import './UserCard.scss';

// Setting material-ui classes
const styles = theme => ({
    user_card: {
        width: '100%',
        height: '50vh',
        display: 'flex',
        alignItems: 'center'
    },
    user_card_content: {
        width: '75%',
        height: '100vh',
        display: 'flex',
        flexDirection: 'column',
        justifyContent: 'center',
        alignItems: 'center'
    },
    user_title: {
        textAlign: 'center'
    },
    typography_media_query: {
        [theme.breakpoints.between('xs', 'sm')]: {
            fontSize: '1.25rem'
        }
    }
});

export const UserCard = (props) => {
    const {
        classes,
        companyData,
        geocoded_address_Origin,
        geocoded_address_Destination,
        userData
    } = props;
    const {
        first_name,
        last_name,
        email,
        phone_number
    } = userData;
    const {
        display_name,
        display_primary_email,
        display_primary_phone_number
    } = companyData.carrier;
    return (
        <Card className={classes.user_card}>
            <CardHeader
                avatar={(
                    <Avatar aria-label="Recipe" className={classes.avatar}>
                    N
                    </Avatar>
                )}
                title={`${first_name} ${last_name}`}
                subheader={`${email} ${phone_number}`}
            />
            <CardContent className={classes.user_card_content}>
                <div className="route">
                    <h4 className="route-info-title">Route</h4>
                    <h5 className="route-info">{`A: ${geocoded_address_Origin}`}</h5>
                    <h5 className="route-info">{`B: ${geocoded_address_Destination}`}</h5>
                    <h4 className="route-info-title">Carrier</h4>
                    <h5 className="route-info">{display_name}</h5>
                    <h5 className="route-info">{display_primary_email}</h5>
                    <h5 className="route-info">{display_primary_phone_number}</h5>
                </div>
            </CardContent>
        </Card>
    );
};

const mapStateToProps = state => ({
    companyData: state.companyData[0],
    geocoded_address_Origin: state.geocoded_address_Origin,
    geocoded_address_Destination: state.geocoded_address_Destination,
    userData: state.userData[0]
});

UserCard.propTypes = {
    classes: PropTypes.object.isRequired,
    companyData: PropTypes.object.isRequired,
    geocoded_address_Origin: PropTypes.string.isRequired,
    geocoded_address_Destination: PropTypes.string.isRequired,
    userData: PropTypes.object.isRequired
};

export default connect(mapStateToProps, null)(withStyles(styles)(UserCard));
