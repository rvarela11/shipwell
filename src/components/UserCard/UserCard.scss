// @styles
@import '../../globals/scss/typography';

.route-info {
    font-weight: 400;
    margin: 0;
}

.route-info-title {
    font-weight: bold;
    margin: rem(10px) 0;
}