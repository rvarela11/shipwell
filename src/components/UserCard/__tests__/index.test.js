// @vendors
import React from 'react';
import { shallow } from 'enzyme';

// @components
import { UserCard, mapStateToProps, styles} from '../UserCard';

describe('UserCard', () => {
    it('renders UserCard without crashing', () => {
        const wrapper = shallow(
            <UserCard
                classes={{}}
                companyData={{ carrier: {} }}
                geocoded_address_Origin=""
                geocoded_address_Destination=""
                userData={{}}
            />
        ).dive();
        expect(wrapper).toMatchSnapshot();
    });
    it('should show initial state from mapStateToProps', () => {
        const initialState = {
            companyData: [{}],
            geocoded_address_Origin: '',
            geocoded_address_Destination: '',
            userData: [{}]
        };
        expect(mapStateToProps(initialState)).toEqual({
            companyData: {},
            geocoded_address_Origin: '',
            geocoded_address_Destination: '',
            userData: {}
        });
    });
});
