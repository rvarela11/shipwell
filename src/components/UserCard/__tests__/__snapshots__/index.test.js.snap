// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserCard renders UserCard without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <UserCard
    classes={Object {}}
    companyData={
      Object {
        "carrier": Object {},
      }
    }
    geocoded_address_Destination=""
    geocoded_address_Origin=""
    userData={Object {}}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <WithStyles(CardHeader)
          avatar={
            <WithStyles(Avatar)
              aria-label="Recipe"
            >
              N
            </WithStyles(Avatar)>
          }
          subheader="undefined undefined"
          title="undefined undefined"
        />,
        <WithStyles(CardContent)>
          <div
            className="route"
          >
            <h4
              className="route-info-title"
            >
              Route
            </h4>
            <h5
              className="route-info"
            >
              A: 
            </h5>
            <h5
              className="route-info"
            >
              B: 
            </h5>
            <h4
              className="route-info-title"
            >
              Carrier
            </h4>
            <h5
              className="route-info"
            />
            <h5
              className="route-info"
            />
            <h5
              className="route-info"
            />
          </div>
        </WithStyles(CardContent)>,
      ],
      "className": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "avatar": <WithStyles(Avatar)
            aria-label="Recipe"
          >
            N
          </WithStyles(Avatar)>,
          "subheader": "undefined undefined",
          "title": "undefined undefined",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <div
            className="route"
          >
            <h4
              className="route-info-title"
            >
              Route
            </h4>
            <h5
              className="route-info"
            >
              A: 
            </h5>
            <h5
              className="route-info"
            >
              B: 
            </h5>
            <h4
              className="route-info-title"
            >
              Carrier
            </h4>
            <h5
              className="route-info"
            />
            <h5
              className="route-info"
            />
            <h5
              className="route-info"
            />
          </div>,
          "className": undefined,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h4
                className="route-info-title"
              >
                Route
              </h4>,
              <h5
                className="route-info"
              >
                A: 
              </h5>,
              <h5
                className="route-info"
              >
                B: 
              </h5>,
              <h4
                className="route-info-title"
              >
                Carrier
              </h4>,
              <h5
                className="route-info"
              />,
              <h5
                className="route-info"
              />,
              <h5
                className="route-info"
              />,
            ],
            "className": "route",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Route",
                "className": "route-info-title",
              },
              "ref": null,
              "rendered": "Route",
              "type": "h4",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "A: ",
                "className": "route-info",
              },
              "ref": null,
              "rendered": "A: ",
              "type": "h5",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "B: ",
                "className": "route-info",
              },
              "ref": null,
              "rendered": "B: ",
              "type": "h5",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Carrier",
                "className": "route-info-title",
              },
              "ref": null,
              "rendered": "Carrier",
              "type": "h4",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
                "className": "route-info",
              },
              "ref": null,
              "rendered": null,
              "type": "h5",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
                "className": "route-info",
              },
              "ref": null,
              "rendered": null,
              "type": "h5",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
                "className": "route-info",
              },
              "ref": null,
              "rendered": null,
              "type": "h5",
            },
          ],
          "type": "div",
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <WithStyles(CardHeader)
            avatar={
              <WithStyles(Avatar)
                aria-label="Recipe"
              >
                N
              </WithStyles(Avatar)>
            }
            subheader="undefined undefined"
            title="undefined undefined"
          />,
          <WithStyles(CardContent)>
            <div
              className="route"
            >
              <h4
                className="route-info-title"
              >
                Route
              </h4>
              <h5
                className="route-info"
              >
                A: 
              </h5>
              <h5
                className="route-info"
              >
                B: 
              </h5>
              <h4
                className="route-info-title"
              >
                Carrier
              </h4>
              <h5
                className="route-info"
              />
              <h5
                className="route-info"
              />
              <h5
                className="route-info"
              />
            </div>
          </WithStyles(CardContent)>,
        ],
        "className": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "avatar": <WithStyles(Avatar)
              aria-label="Recipe"
            >
              N
            </WithStyles(Avatar)>,
            "subheader": "undefined undefined",
            "title": "undefined undefined",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <div
              className="route"
            >
              <h4
                className="route-info-title"
              >
                Route
              </h4>
              <h5
                className="route-info"
              >
                A: 
              </h5>
              <h5
                className="route-info"
              >
                B: 
              </h5>
              <h4
                className="route-info-title"
              >
                Carrier
              </h4>
              <h5
                className="route-info"
              />
              <h5
                className="route-info"
              />
              <h5
                className="route-info"
              />
            </div>,
            "className": undefined,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h4
                  className="route-info-title"
                >
                  Route
                </h4>,
                <h5
                  className="route-info"
                >
                  A: 
                </h5>,
                <h5
                  className="route-info"
                >
                  B: 
                </h5>,
                <h4
                  className="route-info-title"
                >
                  Carrier
                </h4>,
                <h5
                  className="route-info"
                />,
                <h5
                  className="route-info"
                />,
                <h5
                  className="route-info"
                />,
              ],
              "className": "route",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Route",
                  "className": "route-info-title",
                },
                "ref": null,
                "rendered": "Route",
                "type": "h4",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "A: ",
                  "className": "route-info",
                },
                "ref": null,
                "rendered": "A: ",
                "type": "h5",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "B: ",
                  "className": "route-info",
                },
                "ref": null,
                "rendered": "B: ",
                "type": "h5",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Carrier",
                  "className": "route-info-title",
                },
                "ref": null,
                "rendered": "Carrier",
                "type": "h4",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": undefined,
                  "className": "route-info",
                },
                "ref": null,
                "rendered": null,
                "type": "h5",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": undefined,
                  "className": "route-info",
                },
                "ref": null,
                "rendered": null,
                "type": "h5",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": undefined,
                  "className": "route-info",
                },
                "ref": null,
                "rendered": null,
                "type": "h5",
              },
            ],
            "type": "div",
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
